### PR TITLE
yaml-cpp-config.cmake.in: apply patch to fix #774

### LINKED ISSUE #774: Cannot find headers (and libraries) as CMake variables are unset
Hello,

as per issue title, I'm having a problem when I try to use yaml-cpp in a project I'm working on. Namely, the compiler can't find header files (but I suppose the same will go for libraries) when I try to compile. 

This is due to the fact that the file used by CMake to set the paths of include and link directories does not set the proper values for the variables. The content of `share/cmake/yaml-cpp/yaml-cpp-config.cmake` after installation is;:
```
# - Config file for the yaml-cpp package
# It defines the following variables
#  YAML_CPP_INCLUDE_DIR - include directory
#  YAML_CPP_LIBRARIES    - libraries to link against

# Compute paths
get_filename_component(YAML_CPP_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
set(YAML_CPP_INCLUDE_DIR "")

# Our library dependencies (contains definitions for IMPORTED targets)
include("${YAML_CPP_CMAKE_DIR}/yaml-cpp-targets.cmake")

# These are IMPORTED targets created by yaml-cpp-targets.cmake
set(YAML_CPP_LIBRARIES "")
```

As you can see, the variables `YAML_INCLUDE_DIR` and `YAML_CPP_LIBRARIES` are empty. I believe the problem may lie in the input file `yaml-cpp-config.cmake.in`:
```
# - Config file for the yaml-cpp package
# It defines the following variables
#  YAML_CPP_INCLUDE_DIR - include directory
#  YAML_CPP_LIBRARIES    - libraries to link against

# Compute paths
get_filename_component(YAML_CPP_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
set(YAML_CPP_INCLUDE_DIR "@CONFIG_INCLUDE_DIRS@")

# Our library dependencies (contains definitions for IMPORTED targets)
include("${YAML_CPP_CMAKE_DIR}/yaml-cpp-targets.cmake")

# These are IMPORTED targets created by yaml-cpp-targets.cmake
set(YAML_CPP_LIBRARIES "@EXPORT_TARGETS@")
```

Indeed, `grep`-ing for `CONFIG_INCLUDE_DIRS` or `EXPORT_TARGETS` yields no results in the whole repository, so substitution should give an empty string.

Am I missing something?

Thanks!

mattia

### PR DESCRIPTION
**Pull request by Artturin**
From https://github.com/jbeder/yaml-cpp/issues/774#issuecomment-909274732

### CODE REVIEW

**Comment by jbeder** [comment]
Can you add a test that confirms this fixes the issue?

**Comment by Artturin** [comment]
i do not know cpp 

i was updating the rstudio package for nixpkgs and i got this error during configuring

```
CMake Error at src/cpp/CMakeLists.txt:222 (message):
  yaml-cpp not found (re-run dependencies script to install)
```
and applying the patch in this pr to yaml-cpp fixed it 
```
  patches = [
    # https://github.com/jbeder/yaml-cpp/issues/774
    (fetchpatch {
      url = "https://github.com/jbeder/yaml-cpp/commit/4f48727b365962e31451cd91027bd797bc7d2ee7.patch";
      sha256 = "sha256-jarZAh7NgwL3xXzxijDiAQmC/EC2WYfNMkYHEIQBPhM=";
    })
  ];
```

File: yaml-cpp-config.cmake.in [hunk 1]
```diff
@@ -5,10 +5,10 @@
...
 get_filename_component(YAML_CPP_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
-set(YAML_CPP_INCLUDE_DIR "@CONFIG_INCLUDE_DIRS@")
 
 # Our library dependencies (contains definitions for IMPORTED targets)
 include("${YAML_CPP_CMAKE_DIR}/yaml-cpp-targets.cmake")
 
 # These are IMPORTED targets created by yaml-cpp-targets.cmake
-set(YAML_CPP_LIBRARIES "@EXPORT_TARGETS@")
+get_target_property(YAML_CPP_INCLUDE_DIR yaml-cpp INTERFACE_INCLUDE_DIRECTORIES)
+get_target_property(YAML_CPP_LIBRARIES yaml-cpp IMPORTED_LOCATION_RELEASE)
```
Comment (jspricke) [inline yaml-cpp-config.cmake.in:14]:
```suggestion
set(YAML_CPP_LIBRARIES "yaml-cpp")
```
note that this only works with CMAKE_BUILD_TYPE=Release. As the value is static anyhow we could just set it.

**Comment by jbeder** [comment]
It sounds like this is important to get submitted, but I don't understand it. @Artturin, I think what you're trying to fix is an installation step, right?

If so, can you add a case to the github action: https://github.com/jbeder/yaml-cpp/blob/master/.github/workflows/build.yml

If there's a bash command you can run that does the thing you want, that would help ensure this fixes the issue and nothing breaks it down the road.

**Comment by Artturin** [comment]
better solution in https://github.com/jbeder/yaml-cpp/pull/1077

**Pull request closed without merging**